R/unite.R

```diff
@@ -0,0 +1,56 @@
+#' Unite multiple columns into one by pasting strings together.
+#'
+#' @description
+#' This is a method for the tidyr `unite()` generic.
+#'
+#' @inheritParams tidyr::unite
+#' @examples
+#' library(tidyr)
+#'
+#' df <- lazy_dt(expand_grid(x = c("a", NA), y = c("b", NA)))
+#' df
+#'
+#' df %>% unite("z", x:y, remove = FALSE)
+#'
+#' # Separate is almost the complement of unite
+#' df %>%
+#'   unite("xy", x:y) %>%
+#'   separate(xy, c("x", "y"))
+#' # (but note `x` and `y` contain now "NA" not NA)
+# exported onLoad
+unite.dtplyr_step <- function(data, col, ..., sep = "_", remove = TRUE, na.rm = FALSE) {
+  if (is_true(na.rm)) {
+    abort("`na.rm` is not implemented in dtplyr")
+  }
+
+  .col <- as_name(enquo(col))
+
+  dots <- enquos(...)
+  if (length(dots) == 0) {
+    .cols <- data$vars
+  } else {
+    sim_data <- simulate_vars(data)
+    .cols <- names(tidyselect::eval_select(expr(c(!!!dots)), sim_data))
```

Comment:
This should probably use `allow_rename = FALSE`, e.g. something like

```r
locs <- tidyselect::eval_select(expr(c(x, y)), sim_data, allow_rename = FALSE)
.cols <- data$vars[locs]
```

Otherwise, to prevent this error (which also happens for normal dataframes)

```r
df %>% unite("z", c(a = x, y), remove = FALSE)
#> Error in paste(a, y, sep = "_") : object 'a' not found
```

Note that the `allow_rename = FALSE` does not really do anything here, using the locations is the relevant part here. But I think `allow_rename = FALSE` communicates the intent better.